components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -815,5 +801,16 @@ create_dyn_field (const std::string& name,
   m_dyn_grid_fields[name] = f;
 }
 // =========================================================================================
+void HommeDynamics::
+check_computed_fields_impl () {
+    auto& field = m_fields_out["Q"];
+    for (auto& pc : field.get_property_checks()) {
+      if (!pc.check(field) and pc.can_repair()) {
```

Comment:
This is repairing _all_ prop checks that fail, rather than just the one that we know is dynamics fault. Do we want this?

Suggestion: what if we store the FPC in HommeDynamics, and simply run it on `m_fields_out["Q"]`? I would still store a FPC in the field (a FieldWithinIntervalChek, to be precise), without possibility to repair. Then store a second FPC in HommeDynamics (a FieldPositivityCheck) and run it on Q, repairing neg values.

Reply:
It would repair all the ones that "can repair".  But yes, arguably we may want to allow for a different tracer property check that can only be repaired somewhere else.

I like your suggestion of a local FPC that we store in the interface and then run locally on Q instead though.  It's cleaner, and avoids us accidentally repairing something we shouldn't later down the road.

Reply:
Right. The only reason to store FPC's inside Field's, is to allow _other_ atm procs to run the checks. Perhaps only atm proc A knows the correct limits of a field, so it sets up the check, and everyone can then run it.

But in this case, we _don't_ want other atm procs to run this check. It should be a very local quick fix, to work around finite precision issues...